ui/StatusQ/src/StatusQ/Components/StatusImage.qml

```diff
@@ -0,0 +1,20 @@
+import QtQuick 2.13
+Image {
+    id: root
+
+    readonly property bool isLoading: status === Image.Loading
+    readonly property bool isError: status === Image.Error
+
+    fillMode: Image.PreserveAspectFit
+
+    onSourceChanged: {
+        if (sourceSize.width < width || sourceSize.height < height) {
+            sourceSize = Qt.binding(function() {
+                return Qt.size(width * 2, height * 2)
```

Comment:
From StatusIcon:
```qml
    // SVGs must have sourceSize, PNGs not; otherwise blurry
    sourceSize: !!icon ? Qt.size(width, height) : undefined
```

Reply:
Also, wondering why increase in image buffer size. In theory, having one-to-one pixel relation between the screen size and image buffer is ideal.

Reply:
> @stefandunca, good question, not sure why as well, but I remember myself doing the same some years ago, because AFAIR 1:1 relation did not look nice.

Here are some of the reasons why the rendering might not be optimal
- Subpixel alignment: the start of the image is not aligned with a pixel, which might create a blurry image. It is quickly fixed by fixing the alignment. e.g. anchoring to an item with subpixel coordinates (non-`int`) can also be a root cause. Converting to Int a float computation might be a good fix e.g. `toInt(width * 0.5)`
- The texture buffer (rendering buffer) doesn't match and the screen buffer (screen image size); therefore, the buffer is scaled down or up
- Aspect-ratio settings force to stretch the image buffer
- Rendering settings e.g. mipmapping, might improve things

In each case, it is good to fix the root cause and not hide it behind under optimized implementation like this.

Reply:
Hey there! Finally got some time to take a look at this PR again.
@IvanBelyakoff @stefandunca 
I'm not sure why `sourceSize` was calculated this way in the original `StatusRoundedImage` component, and I'm a bit worried about unexpected visual changes if I alter how the sourceSize is calculated, since that one's used all over the app. I'm gonna need some proper time to validate it, so I'll leave it as a separe task if you agree, since this PR has been open long enough as it is.
Also, I cannot use a direct binding for this since the logic involves `sourceSize` itself. I could make it work with a workaround, but I think that's exactly what we're trying to avoid here.

https://github.com/status-im/status-desktop/issues/10139